pkg/omf/functions/packages/omf.packages.install.fish

```diff
@@ -11,44 +11,50 @@ function __omf.packages.install.error.already
 end
 
 function omf.packages.install -a name_or_url
-  if test \( -e $OMF_PATH/db/themes/$name_or_url \) -o (echo $name_or_url | grep theme-)
-    set install_type "theme"
-    set parent_path "themes"
-  else
-    set install_type "package"
-    set parent_path "pkg"
-  end
-
-  if test -e $OMF_PATH/db/$parent_path/$name_or_url
+  if set url (omf.index.stat $name_or_url repository)
     set name $name_or_url
-    set url (cat $OMF_PATH/db/$parent_path/$name)
   else
     set name (omf.packages.name $name_or_url)
     set url $name_or_url
   end
 
-  if test -e $OMF_PATH/$parent_path/$name
-    __omf.packages.install.error.already "$install_type $name_or_url"
+  if contains -- $name (omf.packages.list)
+    __omf.packages.install.error.already "$name_or_url"
     return $OMF_INVALID_ARG
   end
 
-  echo (omf::dim)"Installing $install_type $name"(omf::off)
+  echo (omf::dim)"Installing package $name"(omf::off)
 
-  if omf.repo.clone $url $OMF_PATH/$parent_path/$name
-    omf.bundle.install $OMF_PATH/$parent_path/$name/bundle
-    omf.bundle.add $install_type $name_or_url
+  set -l install_dir $OMF_PATH/pkg/$name
+
+  # Clone the package repository.
+  if not omf.repo.clone $url $install_dir
+    __omf.packages.install.error "$name"
+    return $OMF_UNKNOWN_ERR
+  end
 
-    # Run the install hook.
-    if not omf.packages.run_hook $OMF_PATH/$parent_path/$name install
-      __omf.packages.install.error "$install_type $name"
-      return $OMF_UNKNOWN_ERR
-    end
+  # Check if the package is a theme. If it is, move it to the themes directory.
+  if test -f $install_dir/fish_prompt.fish
```

Comment:
You are totally right about the packages outside the index. For some reason i had in mind that we're also adding a manifest inside each package repository. Though it might make more sense to declare the type inside the package and not in the index, but that's not what this PR is about.
Having said that, I still think it might be better to split this to a separate PR, even just so it's not forgotten when we (probably you...) collect the changes for the next release announcement. Same goes to the removal of `omf submit`, though I'm not sure if anyone cares about it at all.